tests/integ/modin/frame/test_iloc.py

```diff
@@ -3252,3 +3253,47 @@ def ilocset(df):
         native_df,
         ilocset,
     )
+
+
+@pytest.mark.parametrize(
+    "ops",
+    [
+        lambda df: df.head(),
+        lambda df: df.iloc[1:100],
+        lambda df: df.iloc[1000:100:-1],
+    ],
+)
+@sql_count_checker(query_count=6)
+def test_df_iloc_efficient_sql(session, ops):
+    df = DataFrame({"a": [1] * 10000})
+    with session.query_history() as query_listener:
+        ops(df).to_pandas()
+    eval_query = query_listener.queries[
+        -2
+    ].sql_text.lower()  # query before drop temp table
+    # check no row count
+    assert "count" not in eval_query
+    # check orderBy behinds limit
+    assert "count" not in eval_query
+    assert eval_query.index("limit") < eval_query.index("order by")
+
+
+@pytest.mark.parametrize(
+    "ops",
+    [
+        lambda df: df.iloc[0],
+        lambda df: df.iloc[100],
+    ],
+)
+@sql_count_checker(query_count=8, union_count=1)
+def test_df_iloc_scalar_efficient_sql(session, ops):
+    df = DataFrame({"a": [1] * 10000})
+    with session.query_history() as query_listener:
+        ops(df).to_pandas()
+    eval_query = query_listener.queries[
+        -3
+    ].sql_text.lower()  # query before drop temp table and transpose
+    # check no row count
+    assert "count" not in eval_query
+    # check orderBy behinds limit
```

Comment:
this description doesn't seem accurate. There's no search for `orderBy`. Is this an incorrect copy and paste?